### PR TITLE
Academic-Year soft & hard delete functionality API

### PR DESCRIPTION
**Pull request by engrshishir**

### CODE REVIEW

File: database/migrations/2024_04_05_154555_create_academic_years_table.php [hunk 1]
```diff
@@ -14,7 +14,7 @@ public function up(): void
         Schema::create('academic_years', function (Blueprint $table) {
             $table->increments('id');
             $table->string('name', 255);
-            $table->boolean('deleted')->default(false);
+            $table->softDeletes();
```
Comment (ManiruzzamanAkash) [inline database/migrations/2024_04_05_154555_create_academic_years_table.php:17]: Let's keep it as it was previously.

File: routes/api/v1/backend.php [hunk 2]
```diff
@@ -25,4 +25,5 @@
 Route::get('/students/all', [StudentController::class, 'all'])->name('students.all');
 
 // AcademicYear
-Route::apiResource('academicyears', AcademicYearController::class)->only(['index', 'store', 'update', 'destroy']);
\ No newline at end of file
+Route::apiResource('academicyears', AcademicYearController::class)->only(['index', 'store', 'update', 'destroy']);
```
Comment (ManiruzzamanAkash) [inline routes/api/v1/backend.php:28]: `academic-years`

File: app/Services/CrudService.php [hunk 3]
```diff
@@ -56,6 +56,16 @@ public function delete(int $id): object
...
+    public function forceDelete(int $id): object
+    {
+        $entity = $this->repository->findTrash($id);
+        if (!$entity) {
+            $errorMessage = call_user_func($this->generateErrorMessage);
+            throw new NotFoundHttpException($errorMessage);
+        }
+        return $this->repository->forceDelete($entity);
+    }
+
```
Comment (ManiruzzamanAkash) [inline app/Services/CrudService.php:68]: We don't need this method. I've managed CrudService more better way with delete() method.

Considering scenario - 
1. Soft delete
2. Hard Delete
3. Permanently Delete

Comment (ManiruzzamanAkash) [reply app/Services/CrudService.php:68]: Merge main branch in your local and you would get.

**Comment by ManiruzzamanAkash** [comment]
Please take from main branch and see, how handled on Batch routes the soft-delete/hard-delete. 
Maybe need another route also there. But yeah, we'll stick that approach to make next migration easier.

File: app/Services/CrudService.php (hunk 3 above)
Comment (engr-shishir) [reply app/Services/CrudService.php:68]: Vai tahole ki laravel softDelete bad diye dibo?

**Comment by ManiruzzamanAkash** [comment]
Yes, Please remove that.

**Pull request closed without merging**